hhs_facilities/delphi_hhs_facilities/pull.py

```diff
@@ -1,22 +1,59 @@
 """Functions for mapping geographic regions."""
 
+from datetime import date
+
 import pandas as pd
+import numpy as np
+from delphi_utils.geomap import GeoMapper
+from delphi_epidata import Epidata
 
-from .constants import NAN_VALUE
+from .constants import NAN_VALUES
 
 
-def pull_data() -> pd.DataFrame:
+def _pull_data_iteratively(states: set, dates: dict) -> list:
     """
-    Pull HHS data.
+    Pull Epidata API for a set of states and dates.
+
+    To avoid Epidata API row limits, does not grab all values at once. Instead, it loops through
+    each state and pulls all data for 10 hospitals at a time.
 
-    NOTE: Currently these commands are all just for local testing while API is not online yet.
+    Parameters
+    ----------
+    states: set
+      Set of state codes (2 letter lowercase abbreviation) to get data for.
+    dates: dict
+      Dict of 'from' and 'to' dates output by Epidata.range().
+
+    Returns
+    -------
+    List of dictionaries. Concatenation of all the response['epidata'] lists.
+    """
+    responses = []
+    for state in states:
+        lookup_response = Epidata.covid_hosp_facility_lookup(state)
+        state_hospital_ids = [i["hospital_pk"] for i in lookup_response.get("epidata", [])]
+        for i in range(0, len(state_hospital_ids), 50):
+            response = Epidata.covid_hosp_facility(state_hospital_ids[i:i+50], dates)
+            if "too many results, data truncated" in response["message"]:
```

Comment:
it might be safer to check that `response['result'] == 2` (or even `response['result'] != 1`) in case the message ever changes. the status code is intended to be part of the stable api surface, whereas the message is just intended for debugging.

Reply:
I checked -1 originally, but then if a state like virgin islands has no results it throws an error instead of skipping. is `2` == too many results?

Reply:
ah, makes sense.

yeah, `2` is only returned when data is truncated.

Reply:
perfect, I'll have it check for 2 then